File: ios/TestTabApp/ShowPostProps.m
//
//  ShowPostProps.m
//  TestTabApp
//
//  Created by Branding Brand on 2/28/23.
//

#import "ShowPostProps.h"

@implementation ShowPostProps

- (instancetype)initWithComponentName:(NSString *)name data:(NSDictionary *)eventData
{
    self = [self init];
    if (self) {
        _postId = [eventData objectForKey:@"postId"];
        _json = [eventData objectForKey:@"json"];
        _name = [eventData objectForKey:@"name"];
        _dynamicData = [eventData objectForKey:@"dynamicData"];
        _invokingComponent = name;
    }
    return self;
}

- (BOOL)areValidProps
{
    if (self.postId != nil &&
        self.json != nil &&
        self.name != nil &&
        self.invokingComponent != nil) {
        return true;
    }
    else {
        return false;
    }
}

- (NSDictionary *)getJSObjectRepresenation
{
    return @{
        @"id": self.postId,
        @"name": self.name,
        @"json": self.json,
//        @"dynamicData": self.dynamicData,
        @"isLoading": @NO
    };
}
@end

File: ios/TestTabApp/ShowPostProps.swift
//
//  ShowPostProps.swift
//  TestTabApp
//
//  Created by Branding Brand on 2/27/23.
//

import Foundation
import UIKit

class ShowPostProps {
    let id: String?
    let name: String?
    let json: String?
    let discoverPath: String?
    let backButton: Bool?
    let isLoading: Bool?
    let dynamicData: [AnyHashable: Any]?
    let invokingComponent: String?
    
    @objc init(data: Dictionary<AnyHashable, Any>) {
        self.id = data["postId"] as? String
        self.name = data["postId"] as? String
        self.json = data["postId"] as? String
        self.discoverPath = data["postId"] as? String
        self.backButton = data["postId"] as? Bool ?? false
        self.isLoading = data["isLoading"] as? Bool ?? false
        self.dynamicData = data["postId"] as? [AnyHashable: Any]
        self.invokingComponent = data["postId"] as? String
    }
}
